GVFS/GVFS.Service/GVFSService.Mac.cs

```diff
@@ -15,56 +15,88 @@ public class GVFSService
         private const string ServiceNameArgPrefix = "--servicename=";
         private const string EtwArea = nameof(GVFSService);
 
-        private JsonTracer tracer;
+        private ITracer tracer;
         private Thread serviceThread;
         private ManualResetEvent serviceStopped;
         private string serviceName;
-        private RepoRegistry repoRegistry;
+        private IRepoRegistry repoRegistry;
         private RequestHandler requestHandler;
+        private GVFSPlatform gvfsPlatform;
 
-        public GVFSService(JsonTracer tracer)
+        public GVFSService(
+            ITracer tracer,
+            string serviceName,
+            IRepoRegistry repoRegistry,
+            GVFSPlatform gvfsPlatform)
         {
-            string logFilePath = Path.Combine(
-                    GVFSPlatform.Instance.GetDataRootForGVFSComponent(GVFSConstants.Service.ServiceName),
-                    GVFSConstants.Service.LogDirectory);
-            Directory.CreateDirectory(logFilePath);
-
             this.tracer = tracer;
-            this.tracer.AddLogFileEventListener(
-                GVFSEnlistment.GetNewGVFSLogFileName(logFilePath, GVFSConstants.LogFileTypes.Service),
-                EventLevel.Verbose,
-                Keywords.Any);
+            this.repoRegistry = repoRegistry;
+            this.gvfsPlatform = gvfsPlatform;
+            this.serviceName = serviceName;
 
-            this.serviceName = GVFSConstants.Service.ServiceName;
             this.serviceStopped = new ManualResetEvent(false);
             this.serviceThread = new Thread(this.ServiceThreadMain);
-            this.repoRegistry = new RepoRegistry(
-                this.tracer,
-                new PhysicalFileSystem(),
-                GVFSPlatform.Instance.GetDataRootForGVFSComponent(this.serviceName));
-            this.requestHandler = new RequestHandler(this.tracer, EtwArea, this.repoRegistry);
+            this.requestHandler = new RequestHandler(this.tracer, EtwArea, this.repoRegistry);
         }
 
-        public void RunWithArgs(string[] args)
+        public static void CreateAndRun(JsonTracer tracer, string[] args)

```

Comment:
I'm not a fan of creating a static method to do what should be done by the user of the class.  I also don't like that the args of the `Main` are passing down to a class.  I would rather the `Main` method did this work.  Is this possible?  I don't see any tests for this method so it seems like we could.  Otherwise I would only have this method be a `Create` since the run is also public.

Reply:
Moved the creation method to Main.